Rock.Messaging.RabbitMQ/RabbitEngine.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System;
+using System.Threading.Tasks;
+using RabbitMQ.Client;
+using RabbitMQ.Client.Events;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Rock.Messaging.RabbitMQ
+{
+    public class RabbitEngine : IReceiver, ISender
+    {
+        private readonly IRabbitSessionConfiguration _config;
+        private ConnectionFactory _connectionFactory;
+        private IConnection _connection;
+        private IModel _consumerModel; // While this is hacky- IModel is not thread safe.. see: https://www.rabbitmq.com/releases/rabbitmq-dotnet-client/v1.7.1/rabbitmq-dotnet-client-1.7.1-user-guide.pdf
+
+        private bool topical;
+
+        public RabbitEngine(IRabbitSessionConfiguration rabbitSessionConfiguration, bool isTopic = false)
+        {
+            this._config = rabbitSessionConfiguration;
+            topical = isTopic;
+            Initialize();
+        }
+        private void Initialize()
+        {
+            var connectionStrs = _config.QueueUrl.Split(':');
+            _connectionFactory = new ConnectionFactory
+            {
+                UserName = _config.UserName,
+                Password = _config.Password,
+                HostName = connectionStrs[0],
+                Port = int.Parse(connectionStrs[1]),
+                VirtualHost = _config.vHost,
+            };
+            _connection = _connectionFactory.CreateConnection();
+            
+        }
+
+        public void Dispose()
+        {
+            _connection.Dispose();
+            _consumerModel.Abort();
+        }
+
+        #region Receiver
+        public void Start(string selector) // TODO: Discuss: RabbitMQ supports getting individual messages- should Rock.Messaging support that?
```

Comment:
Need to guard against this being called more than once. The second and subsequent times it is called should do nothing.